src/App.tsx

```diff
@@ -15,36 +16,110 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  LENGTH = 'L',
+  ALPHABET = 'A',
+  DEFAULT = '',
+}
+
+function getPreparedGoods(
+  goods: string[],
+  isSorted: string,
+  isReversed: boolean,
+): string[] {
+  const preperedGoods = [...goods];
+
+  switch (isSorted) {
+    case SortType.ALPHABET:
+      preperedGoods.sort((good1, good2) => good1.localeCompare(good2));
+      break;
+    case SortType.LENGTH:
+      preperedGoods.sort((good1, good2) => good1.length - good2.length);
+  }
+
+  if (isReversed) {
+    preperedGoods.reverse();
+  }
+
+  return preperedGoods;
+}
+
 export const App: React.FC = () => {
+  const [selectedSort, setSelectedSort] = useState<string>(SortType.DEFAULT);
+  const [reversed, setReversed] = useState<boolean>(false);
+  const visibleGoods = getPreparedGoods(
+    goodsFromServer,
+    selectedSort,
+    reversed,
+  );
+
+  const onReset = () => {
+    setSelectedSort(SortType.DEFAULT);
+    setReversed(false);
+  };
+
+  const Alphabet = () => {
+    setSelectedSort(SortType.ALPHABET);
+  };
+
+  const Length = () => {
+    setSelectedSort(SortType.LENGTH);
+  };
+
+  const Reversed = () => {
```

Comment:
- Only classes, constructor functions and components should be in PascalCase.
- Function name should start from a verb